Forward tests/Makefile targets to run.d ...

MoonlightSentinel opened the pull request:
... to avoid duplication and missmatches between both test runners.

~~Let's see what breaks ...~~

EDIT: Should be fine now

MoonlightSentinel committed:
Forward tests/Makefile targets to run.d ...

... to avoid duplication and missmatches between both test runners

MoonlightSentinel committed:
run.d: Add CLI switch to dump environment variables

This ensures that the test environment is available in the CI logs.

Geod24 commented:
Instead of `--called-from-make`, why don't you use `.NOTPARALLEL` ? It only works with GNU Make, but so does `-j`, no ?

MoonlightSentinel commented:
> Instead of `--called-from-make`, why don't you use `.NOTPARALLEL` ?

That could work.

> It only works with GNU Make, but so does `-j`, no ?

Yes, but that makefile isn't compatible with DM Make anyway....

MoonlightSentinel commented:
Almost finished but semaphore still fails without an apparent reason (except when tests are run sequentially) ...

MoonlightSentinel committed:
Make gdb sessions mutually exclusive ...

... because SemaphoreCI would abort the tests when multiple
`gdb` tests executed at the same time.

MoonlightSentinel commented:
Seems like `gdb` is the culprit on semaphore because it was terminated after starting mutliple `gdb` tests. It seems to pass with  6c62d5d which ensures that mutliple `gdb` tests are mutually exclusive using a dedicated lockfile.

Geod24 commented on test/run.d:
```diff
@@ -89,14 +89,15 @@ int main(string[] args)
         "f", "Force run (ignore timestamps and always run all tests)", (cast(bool*) &force),
-        "u|unit-tests", "Runs the unit tests", &runUnitTests
+        "u|unit-tests", "Runs the unit tests", &runUnitTests,
+        "e|environment", "Print current environment variables", &dumpEnvironment
```
```suggestion
        "e|environment", "Print current environment variables", &dumpEnvironment,
```

For git-friendliness :)

MoonlightSentinel replied:
Done.

Geod24 reviewed (commented):
Overall LGTM
